rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/BatchJobInfrastructure.java

```diff
@@ -57,662 +57,707 @@
+		if ((jobID = currentNodes.get(nodeName)) != null) {
+			try {
+				deleteJob(jobID);
+			} catch (RMException e) {
+				logger.warn(deleteCmd + " command failed, cannot ensure job " + jobID + " is deleted. Anyway, node "
+						+ nodeName + " is removed from the infrastructure manager.", e);
+			}
+			// atomic remove is important, furthermore we ensure consistent
+			// trace
+			synchronized (currentNodes) {
+				currentNodes.remove(nodeName);
+				logger.debug("Node " + nodeName + " removed. # of current nodes: " + currentNodes.size()
+						+ " # of deploying nodes: " + deployingNodes);
+			}
+		} else {
+			logger.error("Node " + nodeName + " is not known as a Node belonging to this "
+					+ PBSInfrastructure.class.getSimpleName());
+		}
+	}
+
+	/**
+	 * Runs a {@link #getDeleteJobCommand()} command on the remote host for the
+	 * given jobID and monitors the exit.
+	 * 
+	 * @param jobID
+	 *            the jobID string to delete
+	 * @throws RMException
+	 *             if the {@link #getDeleteJobCommand()} command failed
+	 */
+	private void deleteJob(String jobID) throws RMException {
+		String deleteCmd = getDeleteJobCommand();
+		String cmd = deleteCmd + " " + jobID;
+		Process del = null;
+		try {
+			del = Utils.runSSHCommand(InetAddress.getByName(this.serverName), cmd, this.sshOptions);
+		} catch (Exception e1) {
+			logger.warn("Cannot ssh " + this.serverName + " to issue " + deleteCmd + " command. job with jobID: "
+					+ jobID + " won't be deleted.", e1);
+			throw new RMException("Cannot ssh " + this.serverName + " to issue " + deleteCmd
+					+ " command. job with jobID: " + jobID + " won't be deleted.", e1);
+		}
+		long timeStamp = System.currentTimeMillis();
+		while (true) {
+			try {
+				int exitCode = del.exitValue();
+				if (exitCode != 0) {
+					logger.error(
+							"Cannot delete job " + jobID + ". " + deleteCmd + " command returned != 0 -> " + exitCode);
+					throw new RMException(
+							"Cannot delete job " + jobID + ". " + deleteCmd + " command returned != 0 -> " + exitCode);
+				} else {
+					logger.debug("Job " + jobID + " deleted.");
+					return;
+				}
+			} catch (IllegalThreadStateException e) {
+				// the thread hasn't exited yet... don't eat exception, trace
+				// it...
+				logger.trace("waiting for " + deleteCmd + " exit code.", e);
+			}
+			try {
+				Thread.sleep(1000);
+			} catch (InterruptedException e) {
+				// the thread was interrupted... don't eat exception, trace
+				// it...
+				logger.trace("sleep interrupted while waiting for " + deleteCmd + " to exit.", e);
+			}
+			if ((System.currentTimeMillis() - timeStamp) >= nodeTimeOut) {
+				logger.error("Cannot delete job " + jobID + ". " + deleteCmd + " command timed out.");
+				throw new RMException("Cannot delete job " + jobID + ". " + deleteCmd + " command timed out.");
+			}
+		}
+	}
+
+	/**
+	 * @return short description of the IM
+	 */
+	public String getDescription() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add the "@Override" annotation above this method signature [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1161)